### PR TITLE
BIRMINGHAM | ITP-JAN 26 | Merve Reis | sprint 3 | Alarm clock 

### PR DESCRIPTION
**Pull request by mervereis**
## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

- setAlarm function created and wrote the alarmclock app
- Added input validation to handle empty, invalid, and negative values
- Prevented multiple timers by clearing existing intervals before starting a new one
- Cached the `timeRemaining` DOM element to avoid repeated queries
- Handled decimal inputs by rounding down to the nearest second
- Added tests for edge cases (invalid input, negative values, decimals)

### CODE REVIEW

File: Sprint-3/alarmclock/alarmclock.js
Comment (hkavalikas) [inline Sprint-3/alarmclock/alarmclock.js:1]: What will happen if we pass an empty or negative argument as the input?

Comment (hkavalikas) [inline Sprint-3/alarmclock/alarmclock.js:1]: What will happen if I click on `Set alarm` multiple times? How can we improve that?

File: Sprint-3/alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,4 +1,39 @@
...
+
+  return mm + ":" + ss;
+}
+
+function setAlarm() {
+  // Get the number the user typed in the input field
+  let seconds = Number(document.getElementById("alarmSet").value);
+
+  // Update the heading to show the starting time
+  document.getElementById("timeRemaining").innerText =
```
Comment (hkavalikas) [inline Sprint-3/alarmclock/alarmclock.js]: We are using this DOM query in a few places, how can we improve/reuse it instead?

**Review by hkavalikas — CHANGES_REQUESTED** [review]

**Comment by hkavalikas** [comment]
Can we add a Changelist in this PR noting down the changes made?

File: Sprint-3/alarmclock/alarmclock.js
Comment (mervereis) [reply Sprint-3/alarmclock/alarmclock.js:1]: @hkavalikas I added validation to prevent empty, invalid, or negative inputs from starting the timer.

Comment (mervereis) [reply Sprint-3/alarmclock/alarmclock.js:1]: @hkavalikas I now clear any existing interval before starting a new one so multiple clicks don’t create overlapping timers.

File: Sprint-3/alarmclock/alarmclock.js (hunk 1 above)
Comment (mervereis) [reply Sprint-3/alarmclock/alarmclock.js]: @hkavalikas I assigned the timeRemaining element to a variable and reused it for avoid querying the DOM multiple times and improve code reuse.

**Comment by mervereis** [comment]
> Can we add a Changelist in this PR noting down the changes made?

Added a changelist to pr description

**Comment by illicitonion** [comment]
Closing PR because the January ITP run has finished. Feel free to re-open if you're still working on it.
